### PR TITLE
15582 file upload btn kybd focus fix

### PR DESCRIPTION
**Pull request by tlei123**
## Description
[15582](/department-of-veterans-affairs/vets.gov-team/issues/15582) - Restore keyboard-focus support after forcing focus onto an element.

## Testing done
Verified fix locally in browser using keyboard to change focus.
Existing unit-tests still passing.

## Screenshots
See connected ticket for details.

## Acceptance criteria
- [ ] After uploading a file and keyboard-pressing **Delete file** link, keyboard-tabbing focus away from & back to **Upload a document** button works.

## Definition of done
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/platform/forms-system/src/js/utilities/ui/index.js [hunk 1]
```diff
@@ -6,12 +6,20 @@ export function focusElement(selectorOrElement, options) {
     typeof selectorOrElement === 'string'
       ? document.querySelector(selectorOrElement)
       : selectorOrElement;
+  let prevTabIdx = null;
 
   if (el) {
-    if (el.tabIndex <= 0) {
+    prevTabIdx = el.getAttribute('tabindex');
```
Comment (erikphansen) [inline src/platform/forms-system/src/js/utilities/ui/index.js]: just curious if `el.getAttribute('tabindex')` is functionally different than `el.tabIndex`

File: src/platform/forms-system/src/js/utilities/ui/index.js [hunk 2]
```diff
@@ -6,12 +6,20 @@ export function focusElement(selectorOrElement, options) {
...
+    if (prevTabIdx && prevTabIdx <= 0) {
```
Comment (erikphansen) [inline src/platform/forms-system/src/js/utilities/ui/index.js]: I know this is basically existing code but I'm wondering what the point of this check is. Set the element's tabindex to -1 if it's already -1?

**Review by erikphansen — COMMENTED** [review]
Looks fine to me. But am gonna try to get some other folks who have more experience with this existing code to review it.

File: src/platform/forms-system/src/js/utilities/ui/index.js (hunk 2 above)
Comment (jbalboni) [reply src/platform/forms-system/src/js/utilities/ui/index.js]: I think it's related to IE11, but I don't remember exactly.

**Comment by jbalboni** [comment]
@tlei123 @erikphansen Couple things:

1. Make sure to test this in IE11, if you haven't.
2. Can we make these same changes in https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/platform/utilities/ui/index.js as well? This forms-system folder has some duplicate code from the move back from USFS, which we have a ticket to clean up.

**Review by 1Copenut — APPROVED** [review]
@tzelei @erikphansen This looks good to me. I verified the basic functioning with MacOS Chrome. I second what Jeff was saying about testing in IE11 -- I'm not able to log in to test user accounts on my Windows box locally, but can help test IE when you're in Staging. Let me know if you need an extra set of eyes at that time.

**Comment by tlei123** [comment]
@tmwolfe1 , @1Copenut said you may be able to validate this in his absence, against IE11 on Staging.

Please add a comment either with validation results or if you cannot validate.

Thx much!

**Comment by tmwolfe1** [comment]
I will be able to work on this and validate it on Staging (if not today, then by end of day tomorrow).

**Comment by tmwolfe1** [comment]
Able to maintain keyboard focusability on IE 11 after uploading and deleting documents on HCA DD214 (the look in Developer tools is different however on IE 11; appears the value is still 0 for tab index for this button on DOM Explorer, but might need another engineer with more familiarity with IE developer tools to confirm, as I'm not sure this is where I should be checking this there). But it able to continue using the keyboard following upload and deletion of files on IE 11.

**Pull request merged**